qemu/tests/cpu_model.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from virttest import error_context
+from virttest import env_process
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu reboot test:
+    1) Start qemu to get cpu model supported by host
+    3) Boot guest with the cpu model
+    4) Check cpu model name in guest
+    5) Check cpu flags in guest(only for linux guest)
+    6) Reboot guest
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    error_context.context("Start qemu to get support cpu model", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    out = vm.monitor.info("cpu-definitions")
+    vm.destroy()
+    model = params["model"]
+    model_pattern = params["model_pattern"]
+    flags = params["flags"]
+    if utils_misc.get_cpu_vendor(verbose=False) == 'GenuineIntel':
+        model_ib = "%s-IBRS" % model
+        flag_ib = " ibpb ibrs"
+        name_ib = ", IBRS"
+    else:
+        model_ib = "%s-IBPB" % model
+        flag_ib = " ibpb"
+        name_ib = "(with IBPB)"
+
+    model_list = list(filter(lambda x: not x["unavailable-features"], out))
+    models = list(map(lambda x: x["name"], model_list))
+    if model_ib in models:
+        cpu_model = model_ib
+        guest_model = model_pattern % name_ib
+        flags += flag_ib
+    elif model in models:
+        cpu_model = model
+        guest_model = model_pattern % ""
+    else:
+        test.cancel("This host doesn't support cpu model %s" % model)
+
+    params["start_vm"] = "yes"
+    params["auto_cpu_model"] = "no"
+    params["paused_after_start_vm"] = "no"
+    params["cpu_model"] = cpu_model
+
+    env_process.preprocess_vm(test, params, env, params.get("main_vm"))
```

Comment:
I suppose the main purpose here is to change cpu model then reboot the vm, right? 
If so, there is no need to go through `preprocess_vm`.

Reply:
Yes, need change cpu model and restart guest. Can you explain how to do that without preprocess_vm?

Reply:
call `vm.create`, please have a try.

Reply:
Thanks, it works.